nextest-runner/src/list/test_list.rs

```diff
@@ -59,6 +60,9 @@ pub struct RustTestArtifact<'g> {
     /// The working directory that this test should be executed in.
     pub cwd: Utf8PathBuf,
 
+    /// The environment variables to set when running the test.
+    pub env: EnvironmentMap,
```

Comment:
I think this should be passed down as runtime context in `TestList::new`, similar to how the target runner is passed down. This is part of the runtime context, not build-time context.